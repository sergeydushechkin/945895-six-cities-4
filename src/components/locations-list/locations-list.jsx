import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {ActionCreator} from "../../reducer.js";

const LocationsList = (props) => {
  const {city, onLocationButtonClick, locations} = props;

  return (
    <ul className="locations__list tabs__list">
      {locations.map((it) => {
        return (
          <li key={it} className="locations__item">
            <a onClick={(evt) => onLocationButtonClick(evt, it)} className={`locations__item-link tabs__item${it === city ? ` tabs__item--active` : ``}`} href="#">
              <span>{it}</span>
            </a>
          </li>
        );
      })}
    </ul>
  );
};

LocationsList.propTypes = {
  locations: PropTypes.arrayOf(PropTypes.string).isRequired,
  onLocationButtonClick: PropTypes.func.isRequired,
  city: PropTypes.string.isRequired,
};

const mapStateToProps = (state) => {
  return {
    city: state.city,
    offers: state.offers,
    locations: state.locations
  };
};

const mapDispatchToProps = (dispatch) => ({
  onLocationButtonClick(evt, city) {
    evt.preventDefault();
    dispatch(ActionCreator.changeCity(city));
  }
});

export {LocationsList};
export default connect(mapStateToProps, mapDispatchToProps)(LocationsList);
