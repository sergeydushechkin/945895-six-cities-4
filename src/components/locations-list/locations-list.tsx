import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";

import {ActionCreator} from "../../reducer/data/data.js";
import {getLocations, getCity} from "../../reducer/data/selectors.js";

class LocationsList extends React.PureComponent {
  render() {
    const {locations, onCityChange, city} = this.props;

    return (
      <ul className="locations__list tabs__list">
        {locations.map((it) => {
          return (
            <li key={it} className="locations__item">
              <a
                onClick={(evt) => {
                  evt.preventDefault();
                  onCityChange(it);
                }}
                className={`locations__item-link tabs__item${it === city ? ` tabs__item--active` : ``}`}
                href="#"
              >
                <span>{it}</span>
              </a>
            </li>
          );
        })}
      </ul>
    );
  }
}

LocationsList.propTypes = {
  locations: PropTypes.arrayOf(PropTypes.string).isRequired,
  city: PropTypes.string.isRequired,
  onCityChange: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => {
  return {
    city: getCity(state),
    locations: getLocations(state),
  };
};

const mapDispatchToProps = (dispatch) => ({
  onCityChange(city) {
    dispatch(ActionCreator.changeCity(city));
  }
});

export {LocationsList};
export default connect(mapStateToProps, mapDispatchToProps)(LocationsList);
