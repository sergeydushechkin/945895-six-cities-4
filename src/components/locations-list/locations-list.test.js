import React from "react";
import renderer from "react-test-renderer";
import {LocationsList} from "./locations-list.jsx";

it(`Render LocationsList`, () => {
  const tree = renderer
    .create(
        <LocationsList
          locations={[`Amsterdam`, `Paris`, `London`]}
          city={`Paris`}
          onLocationButtonClick={() => {}}
        />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

