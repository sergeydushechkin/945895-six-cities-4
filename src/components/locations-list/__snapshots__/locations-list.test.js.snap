// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render LocationsList 1`] = `
<ul
  className="locations__list tabs__list"
>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        Amsterdam
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item tabs__item--active"
      href="#"
      onClick={[Function]}
    >
      <span>
        Paris
      </span>
    </a>
  </li>
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        London
      </span>
    </a>
  </li>
</ul>
`;
